### PR TITLE
fix context and missing vcHolder implementations

### PR DESCRIPTION
**Pull request by PatStLouis**

### CODE REVIEW

**Review by BigBlueHat — APPROVED** [review]
I like the simplification! I would like skip messages if possible, but merge this first, if you want.

File: tests/95-functions-sd.js [hunk 1]
```diff
@@ -102,23 +110,26 @@ describe('Functions - ecdsa-sd-2023', function() {
...
+          invalidBaseCredential.proof.proofValue =
+            invalidBaseCredential.proof.proofValue.slice(1);
+          ({disclosedCredential} = await createDisclosedVc(
+            {
+              selectivePointers: ['/credentialSubject/id'],
+              signedCredential: invalidBaseCredential,
+              vcHolder: holder
+            }));
+          should.not.exist(disclosedCredential?.proof,
+            'Derive endpoint should reject proof without multibase indicator.'
```
Comment (TallTed) [inline tests/95-functions-sd.js:128]:
```suggestion
            'Derived endpoint should reject proof without multibase indicator.'
```

File: tests/95-functions-sd.js [hunk 2]
```diff
@@ -127,41 +138,48 @@ describe('Functions - ecdsa-sd-2023', function() {
...
+            invalidBaseCredential.proof.proofValue.slice(0, 1) +
+            invalidBaseCredential.proof.proofValue.slice(4);
+          ({disclosedCredential} = await createDisclosedVc(
+            {
+              selectivePointers: ['/credentialSubject/id'],
+              signedCredential: invalidBaseCredential,
+              vcHolder: holder
+            }));
+          should.not.exist(disclosedCredential?.proof,
+            'Derive endpoint should reject proof without header.'
```
Comment (TallTed) [inline tests/95-functions-sd.js:157]:
```suggestion
            'Derived endpoint should reject proof without header.'
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Tiny things

**Comment by PatStLouis** [comment]
@TallTed Thank you for the edits/suggestion, I've committed them. Please approve the corrections so we can merge this PR.

File: tests/95-functions-sd.js [hunk 3]
```diff
@@ -127,41 +138,48 @@ describe('Functions - ecdsa-sd-2023', function() {
...
+            invalidBaseCredential.proof.proofValue.slice(0, 1) +
+            invalidBaseCredential.proof.proofValue.slice(4);
+          ({disclosedCredential} = await createDisclosedVc(
+            {
+              selectivePointers: ['/credentialSubject/id'],
+              signedCredential: invalidBaseCredential,
+              vcHolder: holder
+            }));
+          should.not.exist(disclosedCredential?.proof,
+            'Derived endpoint should reject proof without header.'
```
Comment (dlongley) [inline tests/95-functions-sd.js:157]:
```suggestion
            'Derive endpoint should reject proof without header.'
```

File: tests/95-functions-sd.js [hunk 4]
```diff
@@ -102,23 +110,26 @@ describe('Functions - ecdsa-sd-2023', function() {
...
+          invalidBaseCredential.proof.proofValue =
+            invalidBaseCredential.proof.proofValue.slice(1);
+          ({disclosedCredential} = await createDisclosedVc(
+            {
+              selectivePointers: ['/credentialSubject/id'],
+              signedCredential: invalidBaseCredential,
+              vcHolder: holder
+            }));
+          should.not.exist(disclosedCredential?.proof,
+            'Derived endpoint should reject proof without multibase indicator.'
```
Comment (dlongley) [inline tests/95-functions-sd.js:128]: This is the `/derive` endpoint:

```suggestion
            'Derive endpoint should reject proof without multibase indicator.'
```

**Comment by PatStLouis** [comment]
@TallTed since @dlongley pointed out the the word "Derive" was referring to the "Derive endpoint", the past tense doesn't apply since this is now a noun designating a thing. I've had to revert the suggestions. Could you approve this please?
https://w3c-ccg.github.io/vc-api/#derive-credential

**Review by dlongley — APPROVED** [review]

File: tests/95-functions-sd.js (hunk 1 above)
Comment (TallTed) [inline tests/95-functions-sd.js:128]:
```suggestion
            '"Derive" endpoint should reject proof without multibase indicator.'
```

File: tests/95-functions-sd.js (hunk 2 above)
Comment (TallTed) [inline tests/95-functions-sd.js:157]:
```suggestion
            '"Derive" endpoint should reject proof without header.'
```

**Review by msporny — APPROVED** [review]

**Review by TallTed — APPROVED** [review]

**Pull request merged**